tsdb/head_bench_test.go

```diff
@@ -79,6 +86,71 @@ func BenchmarkHeadStripeSeriesCreate_PreCreationFailure(b *testing.B) {
 	}
 }
 
+func BenchmarkHead_WalCommitWithVaryingSeriesSamplesExemplarsAndHistograms(b *testing.B) {
+	minute := func(m int) int64 { return int64(m) * time.Minute.Milliseconds() }
+	seriesCounts := []int{100, 1000, 10000}
+	series := genSeries(10000, 10, 0, 0)
+	histograms := genHistogramSeries(10000, 10, minute(0), minute(119), minute(1), true)
+
+	for _, seriesCount := range seriesCounts {
+		b.Run(fmt.Sprintf("%d series", seriesCount), func(b *testing.B) {
+			for _, samplesPerAppend := range []int64{1, 2, 5, 100} {
+				b.Run(fmt.Sprintf("%d samples per append", samplesPerAppend), func(b *testing.B) {
+					h, _ := newTestHead(b, 10000, wlog.CompressionNone, false)
+					b.Cleanup(func() { require.NoError(b, h.Close()) })
+
+					ts := int64(1000)
+					appendSamples := func() error {
+						var err error
+						app := h.Appender(context.Background())
+						for _, s := range series[:seriesCount] {
+							var ref storage.SeriesRef
+							for sampleIndex := int64(0); sampleIndex < samplesPerAppend; sampleIndex++ {
+								ref, err = app.Append(ref, s.Labels(), ts+sampleIndex, float64(ts+sampleIndex))
+								if err != nil {
+									return err
+								}
+							}
+						}
+
+						for _, s := range histograms[:seriesCount] {
+							var ref storage.SeriesRef
+							for sampleIndex := int64(0); sampleIndex < samplesPerAppend; sampleIndex++ {
+								ref, err = app.Append(ref, s.Labels(), ts+sampleIndex, float64(ts+sampleIndex))
```

Comment:
```suggestion
								ref, err = app.AppendHistogram(ref, s.Labels(), ts+sampleIndex, float64(ts+sampleIndex))
```

`AppendHistogram` is the method used for histograms, you'll notice that the signature is quite different 😬